prompt_templates/zero_shot/zero_shot_prompt.py

```diff
@@ -0,0 +1,72 @@
+from pydantic import create_model
+import re
+
+
+class ZeroShotPrompt:
+    """
+    A class representing a text template with variable placeholders that can be replaced with actual values.
+
+    Attributes:
+        template (str): The template string with variable placeholders.
+
+    Methods: fill_variables(**variables): Fills the template placeholders with the provided variables and returns the
+    filled text.
+    """
+
+    def __init__(self, template: str):
+        self._template = template
+        self._variables_model = self._infer_variables_model()
+
+    @property
+    def template(self):
+        """
+        Gets the template string with variable placeholders.
+
+        Returns:
+            str: The template string.
+        """
+        return self._template
+
+    def _infer_variables_model(self):
+        variables = self._extract_template_variables()
+        model_fields = {var: (str, ...) for var in variables}
+        variables_model = create_model('TemplateVariables', **model_fields)
```

Comment:
What is a TemplateVariables model - is it a dynamic model that encloses an arbitrary number of variables? If that is the case then this is a cool way of using it. So there is nothing to change here.

Reply:
exactly!